pkg/deploy/api/validation/validation.go

```diff
@@ -37,7 +37,7 @@ func ValidateDeploymentConfig(config *deployapi.DeploymentConfig) fielderrors.Va
 	if config.Spec.Replicas < 0 {
 		allErrs = append(allErrs, fielderrors.NewFieldInvalid("spec.replicas", config.Spec.Replicas, "replicas cannot be negative"))
 	}
-	if config.Spec.Selector == nil || len(config.Spec.Selector) == 0 {
+	if len(config.Spec.Selector) == 0 {
```

Comment:
This predates the upstream discussion to let users customize their
labels.  I think we want to steer towards that as we bring the two
deployment objects in line - it doesn't have to be now (since there's
a lot of backcompat work that has to be done).